test/tests/functional/pbs_ralter.py

```diff
@@ -1108,3 +1109,153 @@ def test_conflict_standing_resv_occurrence(self):
 
         self.alter_a_reservation(rid, start, end, shift, alter_e=True,
                                  whichMessage=0)
+
+    def test_alter_advance_resv_boundary_values(self):
+        """
+        This test checks the alter of start and end times at the boundary
+        values for advance reservation
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4")
+
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_e=True, sequence=1)
+        start, end = self.alter_a_reservation(
+            rid, start, end, -shift, alter_e=True, sequence=2)
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_s=True, sequence=3)
+        self.alter_a_reservation(
+            rid, start, end, -shift, alter_s=True, sequence=4)
+
+    def test_alter_standing_resv_boundary_values(self):
+        """
+        This test checks the alter of start and end times at the boundary
+        values for standing reservation
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid, start, end = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=4", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_e=True, sequence=1)
+        start, end = self.alter_a_reservation(
+            rid, start, end, -shift, alter_e=True, sequence=2)
+        start, end = self.alter_a_reservation(
+            rid, start, end, shift, alter_s=True, sequence=3)
+        self.alter_a_reservation(
+            rid, start, end, -shift, alter_s=True, sequence=4)
+
+    def test_alter_degraded_resv_mom_down(self):
+        """
+        This test checks the alter of start and end times of reservations
+        when mom is down
+        """
+        duration = 30
+        shift = 5
+        offset = 5
+
+        rid1, start1, end1 = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=2")
+        rid2, start2, end2 = self.submit_and_confirm_reservation(
+            offset, duration, select="1:ncpus=2", standing=True,
+            rrule="FREQ=DAILY;COUNT=10")
+        self.mom.stop()
+        msg = 'mom is not down'
+        self.assertFalse(self.mom.isUp(), msg)
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_DEGRADED|10')}
+        self.server.expect(RESV, attrs, id=rid1, max_attempts=30, interval=0.5)
+        self.server.expect(RESV, attrs, id=rid2, max_attempts=30, interval=0.5)
+        self.alter_a_reservation(rid1, start1, end1, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=1)
+        self.alter_a_reservation(rid2, start2, end2, shift, alter_s=True,
+                                 alter_e=True, whichMessage=3, sequence=1)
```

Comment:
From my reading of alter_a_reservation(), whichMessage is pretty much ignored if interactive isn't set.  The CLI returns immediately.

Reply:
No it is now like that.
Since interactive is initialized as 0 by default. So even when "interactive" is not passed in the method when called. It still enters the loop and checks for the message.
`    def alter_a_reservation(self, r, start, end, shift,
                            alter_s=False, alter_e=False,
                            whichMessage=1, confirm=True,
                            interactive=0, sequence=1):` 

`            if interactive:
                if whichMessage == 1:`

I believe you are referring to the above mentioned code

Reply:
interactive defaults to 0.  This means the code won't even check for whichMessage.  It only does that if interactive is set.  So by passing in whichMessage and not passing in interactive, you're basically not checking if the reservation alter is denied.  You're just checking if the reservation alter was requested.